### PR TITLE
Addition of multiple modifiers

### PR DESCRIPTION
**Pull request by addmix**
Really useful modifier.

### CODE REVIEW

**Comment by HungryProton** [comment]
Thanks for the PR. It's rather hard to follow because of all the extra commits though. 
Could you please rebase your master branch so that only the commits that weren't already merged appear here?

A quick summary of what each modifier does could help too. However, I'm not a big fan of splitting the `Apply Offset` modifier in smaller chunks as it breaks backward compatibility and adds unnecessary clutter.

**Comment by addmix** [comment]
> I'm not a big fan of splitting the `Apply Offset` modifier in smaller chunks

The reason I opted to split it up was to allow for user order of operation, such as applying a translational offset relative to the current transform, and then rotating would not be the same as rotating then a local offset. There is a use case for both.

**Comment by addmix** [comment]
I have no fucking clue what I am doing. I am trying to squash the commits but it's only making it worse.

**Comment by addmix** [comment]
After much trial and error, I managed to squash the commits. Hopefully this works out better. I'm still pretty new to contributing on github.

**Comment by HungryProton** [comment]
Thanks for the PR!

The easiest way to handle commits is to create another branch, do your changes there and create the PR from this new branch. It doesn't matter if the commits are not squashed, I have a "Squash and Merge" button that does that automatically on GitHub. Once the PR is merged here, you can update your master branch and delete the old dev branch. 

I can see how useful array and single modifier could be, however apply offset should remain. Eventually we could rename it `Offset Transform` and add three new modifiers: `Offset Position, Offset Rotation and Offset Scale` but in that case, they should be pushed to their own category, I need to fix #31 first.

Array should have a Scale property too in my opinion.

**Comment by addmix** [comment]
> Array should have a Scale property too in my opinion.

I must've overlooked that

**Comment by addmix** [comment]
> we could rename it `Offset Transform`

That sounds like a better way to do it.

I think there may also be an issue with the scale modifier when local is enabled, I need a second set of eyes to look at it though.

**Comment by HungryProton** [comment]
I added a scale property to array and fixed the scaling issue when using global / local. 
Apply Offset is now OffsetTransform and all the offset modifiers were moved in their own category.
